js/Heap.js

```diff
@@ -22,28 +22,75 @@ export default Package.create({
     ios: true,
     android: true,
   }),
-  export: Heap => ({
-    // App Properties
-    setAppId: appId => Heap.setAppId(appId),
-
-    // User Properties
-    identify: identity => Heap.identify(identity),
-    addUserProperties: properties =>
-      Heap.addUserProperties(flatten(properties)),
-
-    // Event Properties
-    addEventProperties: properties =>
-      Heap.addEventProperties(flatten(properties)),
-    removeEventProperty: property => Heap.removeEventProperty(property),
-    clearEventProperties: () => Heap.clearEventProperties(),
-
-    // Events
-    track: (event, payload) => Heap.track(event, flatten(payload)),
-
-    // Redux middleware
-    reduxMiddleware: store => next => action => {
-      Heap.track('Redux Action', flatten(action));
-      next(action);
-    },
-  }),
+  export: Heap => {
+    const track = (event, payload) => {
+      Heap.track(event, flatten(payload));
+    };
+
+    return {
+      // App Properties
+      setAppId: appId => Heap.setAppId(appId),
+
+      // User Properties
+      identify: identity => Heap.identify(identity),
+      addUserProperties: properties =>
+        Heap.addUserProperties(flatten(properties)),
+
+      // Event Properties
+      addEventProperties: properties =>
+        Heap.addEventProperties(flatten(properties)),
+      removeEventProperty: property => Heap.removeEventProperty(property),
+      clearEventProperties: () => Heap.clearEventProperties(),
+
+      // Events
+      track: track,
+
+      // Redux middleware
+      reduxMiddleware: store => next => action => {
+        Heap.track('Redux Action', flatten(action));
+        next(action);
+      },
+
+      autotrackPress: (eventType, componentThis, event) => {
+        const touchableHierarchy = getComponentHierarchy(componentThis._reactInternalFiber);
+        const touchState = componentThis.state.touchable.touchState;
+
+        const eventObj = {
+          touchableHierarchy,
+          touchState,
+        }
+
+        track(eventType, eventObj);
+      },
+    };
+  },
 });
+
+const getComponentHierarchy = (currNode) => {
+  if (currNode === null) {
+    return '';
+  }
+
+  // Skip components we don't care about.
+  // :TODO: (jmtaber129): Skip components with names/display names like 'View' and '_class'.
+  if (!(currNode.elementType !== 'RCTView' && currNode.elementType !== null && (currNode.elementType.displayName || currNode.elementType.name))) {
+    return `${getComponentHierarchy(currNode.return)}`;
+  }
+
+  const elementName = currNode.elementType.displayName || currNode.elementType.name || null;
```

Comment:
Can probably lose the null - we've already checked above that one of these is populated.